### ISSUE TITLE
some geoblocked videos don't work and don't show up in playlists

### ISSUE THREAD

**Issue by lunatherobbers** [iv-org/invidious#2305]
**Describe the bug**
On any instance and no matter what the settings are, it says "The uploader has not made this video available in your country." The video also does not show up in channels or playlists.

**Steps to Reproduce**
Going to any of these example videos:
https://invidious.snopyta.org/watch?v=JEPYvfB_Nrw
https://invidious.snopyta.org/watch?v=jfZe7UHbEXM
https://invidious.snopyta.org/watch?v=VuvfYYlHCv8
https://invidious.snopyta.org/watch?v=hETf2O-QP4E

all of them say "The uploader has not made this video available in your country."
it does not seem to matter which instance it is or whether or not the the proxy option is checked.

and they do not show up in playlists that they're in:

https://invidious.snopyta.org/playlist?list=PLubVgegS36EN8njDO7tyjz_2hxoqAGOmI
https://invidious.snopyta.org/playlist?list=PLRhC9EroPISmsI7cSCWgrf0mKJrfGqUeU
https://invidious.snopyta.org/playlist?list=PLrknD2SRMNPlrh8nyT0QH95Hpt-Kn-QeK
https://invidious.snopyta.org/playlist?list=PLC8GlM8JtFN64eCdqwszCtFTZn2KJfDOn

in fact, all of the geoblocked videos in those playlist have the same issue.

and when the all of the videos in a playlist are geoblocked, it shows the following error,

Title: `Missing hash key: "runs" (KeyError)`
Date: `2021-08-06T06:19:45Z`
Route: `/playlist?list=PLubVgegS36EN8njDO7tyjz_2hxoqAGOmI`
Version: `2021.08.03-e9add69 @ master`
<details>
<summary>Backtrace</summary>
<p>
   
```
Missing hash key: "runs" (KeyError)
  from /usr/share/crystal/src/hash.cr:1027:11 in '[]'
  from /usr/share/crystal/src/json/any.cr:102:7 in '[]'
  from src/invidious/playlists.cr:372:41 in 'fetch_playlist'
  from src/invidious/playlists.cr:355:12 in 'get_playlist'
  from src/invidious/playlists.cr:347:1 in 'get_playlist'
  from src/invidious/routes/playlists.cr:431:18 in 'show'
  from src/invidious.cr:333:1 in '->'
  from /usr/share/crystal/src/primitives.cr:255:3 in '->'
  from /usr/share/crystal/src/primitives.cr:255:3 in 'process_request'
  from lib/kemal/src/kemal/route_handler.cr:17:7 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from lib/kemal/src/kemal/websocket_handler.cr:13:14 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from lib/kemal/src/kemal/filter_handler.cr:21:7 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from src/invidious/helpers/handlers.cr:212:5 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from src/invidious/helpers/handlers.cr:94:12 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from src/invidious/helpers/handlers.cr:145:12 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from src/invidious/helpers/handlers.cr:70:5 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from src/invidious/helpers/static_file_handler.cr:189:11 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from lib/kemal/src/kemal/exception_handler.cr:8:7 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from src/invidious/helpers/logger.cr:17:35 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from lib/kemal/src/kemal/init_handler.cr:12:7 in 'call'
  from /usr/share/crystal/src/http/server/request_processor.cr:51:11 in 'process'
  from /usr/share/crystal/src/http/server.cr:513:5 in 'handle_client'
  from /usr/share/crystal/src/http/server.cr:468:13 in '->'
  from /usr/share/crystal/src/primitives.cr:255:3 in 'run'
  from /usr/share/crystal/src/fiber.cr:92:34 in '->'
  from ???
```
</p>
</details>

**Logs**

**Screenshots**

**Additional context**

**Comment by syeopite** [comment]
This is very interesting! Looks like we finally have an reproducible case of #2278 —  or at least on snoptya!

> all of them say "The uploader has not made this video available in your country."

Yep known issue. We've been having some problems with geoblocked videos. I couldn't track down an exact issue for it though, so I'd keep this open for now.

> and they do not show up in playlists that they're in:

See #2303

**Comment by lunatherobbers** [comment]
upon further investigation, it seems to depend on the country the instance is in, for example

on the one uk instance,
https://invidious.noho.st/watch?v=JEPYvfB_Nrw
it doesn't give that error, but the video still doesn't seem to play
it says "The media could not be loaded, either because the server or network failed or because the format is not supported."

same with a german instance,
https://invidio.xamh.de/watch?v=k-8zNqI42ew

except the video seems to play

but it errors with a US based instance
https://vid.puffyan.us/watch?v=k-8zNqI42ew

so, it seems that if an instance is in a blacklisted country, the videos don't work at all.

but when putting the link into https://www.y2mate.com/youtube/JEPYvfB_Nrw
it allows downloading the video, even when outside of the allowed regions, so I'm not sure what they are doing differently?

**Comment by SamantazFox** [comment]
This should be fixed in #2719.
Regarding the videos not playing, I think that it's because the proxying is broken (or the list of proxies is just outdated).

**Comment by SamantazFox** [comment]
Closing as the problem is likely fixed.
